client/packages/programs/src/JsonForms/components/AdherenceStatus.tsx

```diff
@@ -69,25 +69,15 @@ const UIComponent = (props: ControlProps) => {
 
   // fetch current encounter
   const encounterId = useEncounter.utils.idFromUrl();
-  const { data: currentEncounter, mutate: fetchEncounter } =
-    useEncounter.document.byId(encounterId);
-  useEffect(() => {
-    fetchEncounter();
-  }, [encounterId]);
+  const { data: currentEncounter } = useEncounter.document.byId(encounterId);
 
   // fetch previous encounter
-  const { data: previousEncounter, mutate: fetchPreviousEncounter } =
-    useEncounter.document.previous(
-      currentEncounter?.patient.id ?? '',
-      currentEncounter?.startDatetime
-        ? new Date(currentEncounter?.startDatetime)
-        : new Date()
-    );
-  useEffect(() => {
-    if (currentEncounter) {
-      fetchPreviousEncounter();
-    }
-  }, [currentEncounter]);
+  const { data: previousEncounter } = useEncounter.document.previous(
+    currentEncounter?.patient.id,
```

Comment:
Seems to be working okay -- does it just fail gracefully if currentEncounter is not (yet) defined?

Reply:
The hook disables the query if the patient id is undefined, i.e. if `patient.id === undefined` the query is not run and `previousEncounter` is undefined. Once there is a patient id the query runs and eventually the `previousEncount` variable is set.

I just sneaked in another commit where I just found a similar problem when creating a new patient [c826dda](https://github.com/openmsupply/open-msupply/pull/934/commits/c826ddadeb969677795ced5475886d1366173ad3).